extensions/amp-story/1.0/amp-story-consent.js

```diff
@@ -0,0 +1,275 @@
+                      attrs: dict({'class': 'i-amphtml-story-consent-vendor'}),
+                      children: [],
+                      unlocalizedString: vendor,
+                    })
+                  ),
+                },
+              ],
+            },
+            {
+              tag: 'div',
+              attrs: dict({'class': 'i-amphtml-story-consent-actions'}),
+              children: [
+                {
+                  tag: 'button',
+                  attrs: dict({
+                    'class': 'i-amphtml-story-consent-action ' +
+                        'i-amphtml-story-consent-action-reject',
+                    'on': `tap:${consentId}.reject`,
+                  }),
+                  children: [],
+                  unlocalizedString: 'Decline',
+                },
+                {
+                  tag: 'button',
+                  attrs: dict({
+                    'class': 'i-amphtml-story-consent-action ' +
+                        'i-amphtml-story-consent-action-accept',
+                    'style': config.color ?
+                      `background-color: ${config.color} !important; ` +
+                          `border-color: ${config.color} !important;` : '',
+                    'on': `tap:${consentId}.accept`,
+                  }),
+                  children: [],
+                  unlocalizedString: 'Agree',
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  ],
+});
+
+/**
+ * The <amp-story-consent> custom element.
+ */
+export class AmpStoryConsent extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @const @private {!../../../src/service/action-impl.ActionService} */
+    this.actions_ = Services.actionServiceForDoc(this.element);
+
+    /** @private {?Object} */
+    this.consentConfig_ = null;
+
+    /** @private {?Element} */
+    this.scrollableEl_ = null;
+
+    /** @private {?Element} */
+    this.storyConsentEl_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.assertAndParseConfig_();
+
+    const el = this.win.document.querySelector('[publisher-logo-src]');
+    const logoSrc = el && el.getAttribute('publisher-logo-src');
+
+    if (!logoSrc) {
+      user().warn(
+          TAG, 'Expected "publisher-logo-src" attribute on <amp-story>');
+    }
+
+    const storyConsentConfig =
+        this.consentConfig_ && this.consentConfig_['story-consent'];
+    const consentId = Object.keys(this.consentConfig_.consents)[0];
+    this.storyConsentEl_ = renderAsElement(
+        this.win.document, getTemplate(storyConsentConfig, consentId, logoSrc));
+    createShadowRootWithStyle(this.element, this.storyConsentEl_, CSS);
+
+    this.initializeListeners_();
+    this.addActionsToWhitelist_();
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /** @override */
+  prerenderAllowed() {
+    return true;
```

Comment:
nit: I think we should not prerender this.  It's complicated, but the tl;dr is: We can only pre-render up to 20 AMP components per document, and we gain the most by pre-rendering things that are expensive (i.e. videos) rather than things that are relatively cheap.

Reply:
Done